server/routerlicious/packages/routerlicious-base/src/test/nexus/io.spec.ts

```diff
@@ -338,130 +349,543 @@ describe("Routerlicious", () => {
+								const numberSignalContent = 42;
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									numberSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
+
+								clients.forEach((client, index) => {
 									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
+									);
+									assert.equal(
+										userSignals[index].content,
+										numberSignalContent,
+										`User ${index + 1} signal content mismatch`,
 									);
-									resolve();
 								});
-							}),
-							new Promise<void>((resolve) => {
-								secondSocket.on("signal", (signal: ISignalMessage) => {
+							});
+							it("can handle boolean content", async () => {
+								const booleanSignalContent = true;
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									booleanSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
+
+								clients.forEach((client, index) => {
 									assert.equal(
-										signalContent,
-										signal.content,
-										"Signal content mismatch",
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
 									);
 									assert.equal(
-										firstConnectMessage.clientId,
-										signal.clientId,
-										"Client ID mismatch",
+										userSignals[index].content,
+										booleanSignalContent,
+										`User ${index + 1} signal content mismatch`,
 									);
-									resolve();
 								});
-							}),
-						];
+							});
+							it("can handle JSON/Object content", async () => {
+								const objectSignalContent = {
+									key1: "value1",
+									key2: 42,
+									key3: true,
+								};
+								clients[0].socket.send("submitSignal", clients[0].clientId, [
+									objectSignalContent,
+								]);
+								const userSignals = await Promise.all(
+									clients.map((client) => client.signalPromise),
+								);
 
-						firstSocket.send("submitSignal", firstConnectMessage.clientId, [
-							signalContent,
-						]);
-						await Promise.all(promises);
-					});
+								clients.forEach((client, index) => {
+									assert.equal(
+										client.signalCount,
+										1,
+										`User ${index + 1} should have received 1 signal`,
+									);
+									assert.deepEqual(
+										userSignals[index].content,
+										objectSignalContent,
+										`User ${index + 1} signal content mismatch`,
+									);
+								});
+							});
+							it("can handle malformed JSON content", async () => {
+								const malformedJsonSignalContent =
+									"{key1: value1, key2: 42, key3: true}";
```

Comment:
Is this malformed JSON?

Reply:
I believe so, property keys in JSON have to be enclosed in double quotes. If you pass it to JSON.parse() you will get an error:

on playground: https://www.typescriptlang.org/play?#code/MYewdgzgLgBA5iEATAUgZQPIDkYF4brYB0ADgIYBOEApgBQDkA3gEQDW1AngIzMBcMzAG5kANgFdqPADQD2HAEx8YAFnky2nAMxKoFCQF96ASgDcAKFCQQI6kREg4tBMkJZTF8NBgBbUQDMQCm9qVEwcfFdSShpaZkY5Ln56YXFJehk5eX5VDK1+XQNmdyA

Reply:
But it isn't malformed Json - it is mis-encoded Json.
Does this test stand up to the "you can break it with reasonable bug in implementation" criteria? When would implementation try to parse this? Is should always just be a string to service and never examined, right?